numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -691,15 +691,28 @@ NpyIter_Deallocate(NpyIter *iter)
         }
     }
 
-    /* Deallocate all the dtypes and objects that were iterated */
+    /*
+     * Deallocate all the dtypes and objects that were iterated and resolve
+     * any writeback buffers created by the iterator
+     */
     for(iop = 0; iop < nop; ++iop, ++dtype, ++object) {
+        if (op_itflags[iop] & NPY_OP_ITFLAG_HAS_WRITEBACK) {
```

Comment:
For consistency with how `dtype` and `object` are used below, you could use `*op_itflags` and increment it along with the others.

Probably not worth it though - if anything I find your approach clearer.